PixelGamEngineSolution/FlappyBird/Main.cpp

```diff
@@ -1,76 +1,273 @@
 #define OLC_PGE_APPLICATION
 #include "olcPixelGameEngine.h"
 
+#include "Bird.h"
 
 class FFlappyBird : public olc::PixelGameEngine
 {
 private:
-	float BirdPosition_y = 0.0f;
-	float BirdVelocity_y = 0.0f;
-	float BirdAcceleration_y = 0.0f;
 
 	float Gravity = 100.0f;
 
+	float SectionWidth;
+	std::list<int> ListSection;
+	float LevelPosition = 0.0f;
+
+	bool bHasCollided = false;
+	bool bResetGame = true;
+
+	int MaxFlapCount = 0;
+	int FlapsCount = 0;
+	int AttemptsCount = 0;
+
+	FBird RedBird;
+
+	olc::Sprite* BgSprite = nullptr;
+	olc::Decal* BgDecal = nullptr;
+
+	olc::Sprite* UpSprite = nullptr;
+	olc::Decal* UpDecal = nullptr;
+
+	olc::Sprite* DownSprite = nullptr;
+	olc::Decal* DownDecal = nullptr;
+
+	olc::Sprite* PipeSprite = nullptr;
+	olc::Decal* PipeDecal = nullptr;
+
+	olc::Sprite* OverSprite = nullptr;
+	olc::Decal* OverDecal = nullptr;
+
+	std::vector<olc::Sprite*> NumbersSprites;
+	std::vector<olc::Decal*> NumbersDecal;
 
 public:
 	FFlappyBird()
 	{
-		sAppName = "Example";
+		sAppName = "Flappy Bird";
 	}
 
 protected:
 	bool OnUserCreate() override
 	{
-		// Called once at the start, so create things here
+		ListSection = { 0, 0, 0 };
+		SectionWidth = (float)ScreenWidth() / (float)(ListSection.size() - 1);
+
+		BgSprite = new olc::Sprite("./sprites/background-day.png");
+		BgDecal = new olc::Decal(BgSprite);
+
+		UpSprite = new olc::Sprite("./sprites/redbird-upflap.png");
+		UpDecal = new olc::Decal(UpSprite);
+
+		DownSprite = new olc::Sprite("./sprites/redbird-downflap.png");
+		DownDecal = new olc::Decal(DownSprite);
+
+		PipeSprite = new olc::Sprite("./sprites/pipe-green-down.png");
+		PipeDecal = new olc::Decal(PipeSprite);
+
+		OverSprite = new olc::Sprite("./sprites/gameover.png");
+		OverDecal = new olc::Decal(OverSprite);
+
+		NumbersSprites = {
+			new olc::Sprite("./sprites/0.png"),
+			new olc::Sprite("./sprites/1.png"),
+			new olc::Sprite("./sprites/2.png"),
+			new olc::Sprite("./sprites/3.png"),
+			new olc::Sprite("./sprites/4.png"),
+			new olc::Sprite("./sprites/5.png"),
+			new olc::Sprite("./sprites/6.png"),
+			new olc::Sprite("./sprites/7.png"),
+			new olc::Sprite("./sprites/8.png"),
+			new olc::Sprite("./sprites/9.png"),
+		};
+
+		for (olc::Sprite* InSprite : NumbersSprites)
+		{
+			NumbersDecal.push_back(new olc::Decal(InSprite));
+		}
+
 		return true;
 	}
 
+	// --------------------------------------------------------------------------------
+
 	bool OnUserUpdate(float ElapsedTime) override
 	{
-		// called once per frame
+		if (bResetGame)
+		{
+			ResetGame();
+		}
+
+		Clear(olc::BLACK);
+		DrawDecal(olc::vf2d(0, 0), BgDecal, {0.15f, 0.15f});//Print BackGround
+		SectionThing();
+		DrawBird();
+
+		if (bHasCollided)
+		{
+			DrawDecal(olc::vf2d(ScreenWidth() / 6, ScreenHeight() / 4), OverDecal, { 0.15f, 0.15f });
```

Comment:
o; It's nice you're dealing with the game over with a special screen, however the name OverDecal could be GameOverDecal, it's just 4 plus letters and it says a ton more